src/App.jsx

```diff
@@ -3,46 +3,62 @@ import React, { useState, useEffect } from 'react';
 import AddTodoForm from './AddTodoForm';
 import TodoList from './TodoList';
 
-function useSemiPersistentState() {
-  const [todoList, setTodoList] = useState(
-    JSON.parse(localStorage.getItem('savedTodoList')) || []
-  );
-
-  useEffect(() => {
-    localStorage.setItem('savedTodoList', JSON.stringify(todoList));
-  }, [todoList]);
-
-  return [todoList, setTodoList];
-}
-
 function App() {
-  const [todoList, setTodoList] = useSemiPersistentState();
+  const [todoList, setTodoList] = useState(JSON.parse(localStorage.getItem('todoList')) ?? []);
```

Comment:
I see you are committing to using the "nullish" operator, which is fine but what is your reasoning for this?

Reply:
I wanted to use this operator, according to my criteria it is more precise and specific in this context, but I am going to change it to II to be able to expand other values. `const storedTodoList = JSON.parse(localStorage.getItem('todoList')) || [];`